src/CMakeLists.txt

```
@@ -0,0 +1,79 @@
+add_subdirectory(vmec_interface)
+
+
+add_executable(stella.x  ${SOURCES} )
+
+#set flags here
+#TODO add the most common flags needed for stella
+
+set(dialect "-ffree-line-length-none -fno-backslash -I /usr/include")
+set(bounds "-fbounds-check")
+set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} ${dialect} -march=native")
+set(CMAKE_Fortran_FLAGS_DEBUG "${CMAKE_Fortran_FLAGS_DEBUG} ${bounds} -pedantic -g -Wall -fimplicit-none")
+if(PRECISION STREQUAL "DOUBLE")
+	set(CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fdefault-real-8 -fdefault-double-8")
+endif()
+
+#get package includes and libs here
+
+find_package(MPI REQUIRED)
+if(MPI_FOUND)
+	message(STATUS "Found MPI Libraries in ${MPI_LIBRARIES}")
+	get_filename_component (MPI_LIBRARY_PATH ${MPI_LIBRARY} DIRECTORY)
+    target_include_directories(stella.x PRIVATE ${MPI_Fortran_INCLUDE_PATH})
+	target_link_libraries( stella.x PRIVATE ${MPI_Fortran_LIBRARIES})
```

CMake's targets are great and mean we can just link against the target name, and it automatically sorts out the include directories too:

```suggestion
target_link_libraries( stella.x PRIVATE MPI::MPI_Fortran)
```